### PR TITLE
ContentCredentials CLI fix for stream

### PR DESCRIPTION
**Pull request by damoore044**
### Problem Statement
In Stream runs, for ContentCredentials, we see some occasional failures in several tests, 
that attempt to add/update gpg key associated to repo/product with repo(s), then read info of the product/repo and assert the key name:
```
   ... setup custom repo, GPG key, add key to repo or repo's product,
        or update a GPG key already added to repo(s)

   repo = target_sat.cli.Repository.info({'id': repo['id']}) 
   assert repo['gpg-key'].get('name') == new_name
E   KeyError: 'gpg-key'
```

### Solution
We need to wait for an in-progress _**Repository Metadata Generate task**_, 
that is not handled synchronously by cli `.update` or cli `.info`. Else, the GPG key is not yet present in repo/product.

In any CLI tests adding/updating a GPG key associated to repo, or adding the key to a product containing repos, 
   we should wait for these repo metadata task(s), before calling info of product/repo, and asserting the key.

### Related Issues
- Intermittent Stream CLI failures in ContentCredentials, el8 and el9.
- 1-3 of the affected tests inconsistently fail for same `KeyError`. Locally reproduced when full cli module is run.
- The metadata task is one per repo with the key added, and typically lasts 3 seconds or less, 
but it is enough for an intermittent failure when trying to read the entity (repos or product) and find the key, before the task(s) finished.

### PRT Case
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentcredentials.py
```

### CODE REVIEW

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentcredentials.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7100
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : =========== 1 failed, 83 passed, 216 warnings in 1346.78s (0:22:26) ============
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentcredentials.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7101
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : ============ 7 passed, 116 warnings, 77 errors in 834.60s (0:13:54) ============
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7107
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : ============ 7 passed, 115 warnings, 77 errors in 733.79s (0:12:13) ============
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentcredentials.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7110
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : ============ 7 passed, 115 warnings, 77 errors in 760.52s (0:12:40) ============
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentcredentials.py
nailgun: 1151

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7113
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : ================ 84 passed, 222 warnings in 1270.67s (0:21:10) =================
```

**Review by vsedmik — APPROVED** [review]
Nice catch, ACK!
Non-blocking - the comments are not necessary imho - the function name and docstring are self-describing enough.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7141
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : ================ 84 passed, 223 warnings in 1348.98s (0:22:28) =================
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7143
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : =========== 1 failed, 83 passed, 216 warnings in 1153.64s (0:19:13) ============
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7144
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : =========== 1 failed, 83 passed, 222 warnings in 1196.17s (0:19:56) ============
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentcredentials.py
nailgun: 1152

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentcredentials.py
nailgun: 1152

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7147
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : =========== 1 failed, 83 passed, 230 warnings in 1196.63s (0:19:56) ============
```

**Review by sambible — APPROVED** [review]
Change looks good, only 1 failure being returned seems positive. Still, I'd like to see a consistent pass here before we merge :)

**Comment by shweta83** [comment]
@damoore044 Please make sure the test failing is unrelated to this change. The change looks good otherwise.

**Review by shweta83 — APPROVED** [review]

**Comment by damoore044** [comment]
The remaining failure seems to be the same kind of flakiness in `test_positive_update_key_for_product_with_repos `
will try to address this soon

**Review by ColeHiggins2 — APPROVED** [review]
ack pending prt

**Review by ColeHiggins2 — APPROVED** [review]
ack pending prt

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentcredentials.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7170
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : ================ 84 passed, 237 warnings in 1237.59s (0:20:37) =================
```

File: tests/foreman/cli/test_contentcredentials.py [hunk 1]
```diff
@@ -45,6 +45,27 @@ def create_gpg_key_file(content=None):
     return None
 
 
+def wait_for_repo_metadata_tasks(sat, org_name, repo_name='', product_name=''):
+    """Search and wait for any repository metadata generate task(s).
+
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_contentcredentials.py:50]: Can you please explain the rationale for adding a new function for this instead using existing `wait_for_tasks` helper func directly in tests?

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentcredentials.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7185
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_contentcredentials.py --external-logging
Test Result : ================ 84 passed, 229 warnings in 1235.96s (0:20:35) =================
```

File: tests/foreman/cli/test_contentcredentials.py (hunk 1 above)
Comment (damoore044) [reply tests/foreman/cli/test_contentcredentials.py:50]: They all need pretty much the same thing so saw the opportunity to streamline, 
and perhaps any future use in other CLI tests?

**Pull request merged**